lab_2_spellcheck/main.py

```diff
@@ -72,6 +135,16 @@ def calculate_distance(
 
     In case of corrupt input arguments or unsupported method, None is returned.
     """
+    if not isinstance(first_token, str):
+        return None
+    if not isinstance(vocabulary, dict):
+        return None
+    if not all(isinstance(word, str) for word in vocabulary.keys()):
+        return None
+    if not all(isinstance(word_freq, float) for word_freq in vocabulary.values()):
+        return None
```

Comment:
Use `check_dict` function and don't forget to check corrupt input for other arguments